packages/core/src/request-builder/request/odata-request.ts

```diff
@@ -146,54 +151,99 @@ export class ODataRequest<RequestConfigT extends ODataRequestConfig> {
    * @returns Key-value pairs where the key is the name of a header property and the value is the respective value
    */
   async headers(): Promise<MapType<string>> {
-    return buildHeaders(this).catch(error =>
-      Promise.reject(
+    try {
+      if (!this.destination) {
+        throw Error('The destination is undefined.');
+      }
+      const defaultHeaders = replaceDuplicateKeys(
```

Comment:
why has this been moved from the header-builder to here? Is there no reuse potential?

Reply:
I wanted to get rid of the notion of odata request in the header builder. I don't think there is much potential for reuse because it is solely based on our odata request.